panic: assert failed: series key for ID: not found

Even with #9391, still managed to run into this. After deleting lots of series, it wouldn't start back up (no panic during the actual deletion). Manually rebuilding the indices with `influx_inspect buildtsi` fixed the issue.

```
panic: assert failed: series key for ID: 262444 not found

goroutine 243 [running]:
github.com/influxdata/influxdb/tsdb/index/tsi1.assert(0xdefe00, 0xc4218e27e0, 0x23, 0x0, 0x0, 0x0)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/tsi1.go:531 +0xf6
github.com/influxdata/influxdb/tsdb/index/tsi1.(*LogFile).execSeriesEntry(0xc4216ecfc0, 0xc4204f5c60)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/log_file.go:621 +0x225
github.com/influxdata/influxdb/tsdb/index/tsi1.(*LogFile).execEntry(0xc4216ecfc0, 0xc4204f5c60)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/log_file.go:575 +0x68
github.com/influxdata/influxdb/tsdb/index/tsi1.(*LogFile).open(0xc4216ecfc0, 0xc420e58780, 0xc420e586e0)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/log_file.go:135 +0x3a8
github.com/influxdata/influxdb/tsdb/index/tsi1.(*LogFile).Open(0xc4216ecfc0, 0xc42555e4b0, 0x42)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/log_file.go:83 +0x2b
github.com/influxdata/influxdb/tsdb/index/tsi1.(*Partition).openLogFile(0xc4216ec5a0, 0xc42555e4b0, 0x42, 0xc42555e4b0, 0x42, 0x8)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/partition.go:222 +0x5a
github.com/influxdata/influxdb/tsdb/index/tsi1.(*Partition).Open(0xc4216ec5a0, 0x0, 0x0)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/partition.go:164 +0x5f4
github.com/influxdata/influxdb/tsdb/index/tsi1.(*Index).Open.func1(0xc4266de104, 0x8, 0xc421c06120, 0xc4205e1620, 0x7)
	/go/src/github.com/influxdata/influxdb/tsdb/index/tsi1/index.go:201 +0x30
```

cc @benbjohnson @louisvernon

--- a/tsdb/index/tsi1/log_file.go
+++ b/tsdb/index/tsi1/log_file.go
@@ -618,7 +618,14 @@ func (f *LogFile) execSeriesEntry(e *LogEntry) {
 		seriesKey = f.sfile.SeriesKey(e.SeriesID)
 	}
 
-	assert(seriesKey != nil, fmt.Sprintf("series key for ID: %d not found", e.SeriesID))
+	// Series keys can be removed if the series has been deleted from
+	// the entire database and the server is restarted. This would cause
+	// the log to replay its insert but the key cannot be found.
+	//
+	// https://github.com/influxdata/influxdb/issues/9444
+	if seriesKey == nil {
+		return
+	}
 
 	// Check if deleted.
 	deleted := e.Flag == LogEntrySeriesTombstoneFlag

